src/GPS.cpp

```diff
@@ -0,0 +1,73 @@
+#include "GPS.hpp"
+#include <nmea0183/GPS.hpp>
+
+using namespace nmea0183;
+using namespace marnav;
+using namespace std;
+using namespace gps_base;
+
+GPS::GPS(Driver& driver)
+    : m_driver(driver)
+{
+}
+
+gps_base::GPS_SOLUTION_TYPES GPS::getPositionType(
+    marnav::utils::optional<marnav::nmea::mode_indicator> mode)
+{
+    // TODO: check returns
+    switch (*mode) {
+        case marnav::nmea::mode_indicator::invalid:
+            return GPS_SOLUTION_TYPES::INVALID;
+        case marnav::nmea::mode_indicator::autonomous:
+            return GPS_SOLUTION_TYPES::AUTONOMOUS_2D;
```

Comment:
As I could see in the gps_base documentation, AUTONOMOUS_2D is like AUTONOMOUS, but without the heifht information. RMC messages doesn't output the altitude/height, so I thought it would be more suitable here.

Reply:
![image](https://github.com/user-attachments/assets/0bc54181-05de-4178-a3b6-adde72a53cc2)

Reply:
This is not relevant to the message. It is relevant to the GPS solution. A solution without height is not a full solution (and will usually be wrong by 10s of meters)